lib/src/solver/incompatibility.dart

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import '../package_name.dart';
+import 'term.dart';
+
+/// A set of mutually-incompatible terms.
+///
+/// See https://github.com/dart-lang/pub/tree/master/doc/solver.md#incompatibility.
+class Incompatibility {
+  /// The mutually-incompatibile terms.
+  final List<Term> terms;
+
+  /// Creates an incompatibility with [terms].
+  ///
+  /// This normalized [terms] so that each package has at most one term
+  /// referring to it.
+  factory Incompatibility(List<Term> terms) {
+    if (terms.length == 1 ||
```

Comment:
It *means* `not <term>`, but it can't be *replaced* by it unless we want to have `Term` implement `Incompatibility`, which I think would be pretty confusing.